### ISSUE TITLE
Csharp code in Reference docs

### ISSUE THREAD

**Issue by evilz** [tpetricek/FSharp.Formatting#307]
Hi,

I try to add some csharp code in my reference documentation. But the syntax is not highlighted.

my code : 

```
/// <summary>
/// some blablabla
/// 
///		[lang=csharp]
///		Console.WriteLine("Hello world!");
/// 
/// </summary>
```

Do I make something wrong ?

**Comment by matthid** [comment]
No, this is currently not supported (IIRC) (see https://github.com/tpetricek/FSharp.Formatting/issues/36).

To quote @tpetricek:

> Yes, that's the plan!

I'm pretty sure this should be simple and straight-forward to add, but there wasn't enough interest in that particular feature for someone to jump in. It's just a matter of adding (already existing) markup transformations to the documentation markup.

I will probably jump into this eventually (but feel free to submit a PR in the mean time). My plan is to fix this in combination with https://github.com/tpetricek/FSharp.Formatting/issues/44 (will comment there as well).

**Comment by evilz** [comment]
#44 is very interesting ! I think having tooltip in visual is important.

Have you any plan about generating a different visual for Reference docs for C# or VB ? 
F# signature and tooltip is cool, but I'm not sure every .net developers understand it.

**Comment by matthid** [comment]
> Have you any plan about generating a different visual for Reference docs for C# or VB ? 
F# signature and tooltip is cool, but I'm not sure every .net developers understand it.

Yes I know what you mean. The visual appearance is easily changed with some razor templates. The tool-tip and signature are other stories though... They are generated here: https://github.com/tpetricek/FSharp.Formatting/blob/master/src/FSharp.MetadataFormat/Main.fs#L369
The easiest thing would be to add another optional parameter for the language we want the usages and signatures to have. Maybe we could even extract that part to a razor template. But there aren't any plans for it AFAIK. (And it could become a bit tricky for F# functions with multiple arguments and such things).
Personally I (and probably some people here) like having the F# signature, even when the project is in fact C#. :)

**Comment by tpetricek** [comment]
This is something I wanted to do for quite some time, so thanks for the reminder :-).

I added an initial implementation of this [with this commit](https://github.com/tpetricek/FSharp.Formatting/commit/4de974ff928d34edc3ce390dc8d748056b5fbffa). I suspect there might be a few limitation that we can address later (like, it would be nice to automatically open some namespaces etc.), but basic things should work now.

Adding more flexibility to how the signatures are formatted is a separate issue - it would be nice to have that too.

**Comment by tpetricek** [comment]
Released in v2.9.4.
